Source/Core/DolphinQt2/Config/FilesystemWidget.h

```diff
@@ -7,7 +7,11 @@
 #include <QDialog>
 #include <memory>
 
+// Some versions of moc choke on this header, so we have to ifguard it.
```

Comment:
While this is a good start, it doesn't exactly explain what the specific error is or why it happens.

Reply:
This is likely because of nested-namespace specifiers, by the way. Support for which is added in Qt 5.9 [relevant code review](https://codereview.qt-project.org/#/c/187819/). So whoever reported the above issue is using an old(er) version of Qt.